File: players/python/player.py
#!/bin/env python3
import random

from proboj import *


class MyPlayer(ProbojPlayer):
    def make_turn(self) -> List[Turn]:
        """
        Sem patrí váš kod. Táto metóda by mala vrátiť pole vašich ťahov.
        """
        self.log(self.harbors)
        self.log(self.ships)
        self.log(self.myself, self.myself.gold)
        # self.log(self.map)
        self.log(self.is_occupied_by_ship(XY(0, 0)))

        moves = [BuyTurn(ShipsEnum.Cln)]
        self.log(self.base)
        for ship in self.mine_ships():
            path = Utils.bfs_path(ship.coords, self.harbors[0].coords, self.map)
            if path:
                moves.append(MoveTurn(ship.index, path[0]))
        self.log(moves)

        return moves


if __name__ == "__main__":
    p = MyPlayer()
    p.run()

File: players/cpp/player.cpp
#include "common.h"
#include "proboj.h"
using namespace std;

World world;

vector<Turn> do_turn(){
    // sem ide vas kod
    vector<Turn> turns;
    for(Ship ship : world.my_ships()){
        if(ship.coords == world.my_base && ship.resources[ResourceEnum::Gold] == 0)
            turns.push_back(StoreTurn(ship.index,-5));
        else
			// takto sa pohybuje
            turns.push_back(MoveTurn(ship.index,{50,50}));
    }
    if(world.my_ships.size() < 3)
		// takto mozes kupit lod
        turns.push_back(BuyTurn(ShipsEnum::Cln))
    cerr << "Takto mozete vypisovat do logov" << endl;
    cerr << turns << endl;
    return turns;
}


int main(){
    char dot;
    while(1){
        cin >> world >> dot;
        for(Turn turn : do_turn())
            cout << turn << "\n";
        cout << dot << endl;
    }
}
